swap_meet/vendor.py

```diff
@@ -1,2 +1,58 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+
+    def add(self, new_item):
+        self.inventory.append(new_item)
+        return new_item
+
+    def remove(self, item_to_remove):
+        if item_to_remove in self.inventory:
+            removed_item = self.inventory.pop(self.inventory.index(item_to_remove))
+            return removed_item
+        return False
+
+    def get_by_category(self, category):
+        category_items = []
+        for item in self.inventory:
+            if item.category == category:
+                category_items.append(item)
+        return category_items
```

Comment:
Great logic! As an alternative, can you think how you would do this as a list comprehension? Which style do you prefer?